lib/process-services-cloud/src/lib/people/components/people-cloud.component.ts

```diff
@@ -174,62 +196,56 @@ export class PeopleCloudComponent implements OnInit, OnChanges, OnDestroy {
-                    this.searchUserCtrl.markAsUntouched();
-                }
-            }),
-            tap(() => {
-                this.resetSearchUsers();
-            }),
-            switchMap((search) =>
-                this.identityUserService.findUsersByName(search.trim())),
-            mergeMap((users) => {
-                this.resetSearchUsers();
-                this.searchLoading = false;
-                return users;
-            }),
-            filter(user => !this.isUserAlreadySelected(user)),
-            mergeMap(user => {
-                if (this.appName) {
-                    return this.checkUserHasAccess(user.id).pipe(
-                        mergeMap(
-                            hasRole => hasRole ? of(user) : of()
-                        )
-                    );
-                } else if (this.hasRoles()) {
-                    return this.filterUsersByRoles(user);
-                } else {
-                    return of(user);
-                }
-            }),
-            takeUntil(this.onDestroy$)
-        ).subscribe(user => {
-            this._searchUsers.push(user);
-            this.searchUsers$.next(this._searchUsers);
-        });
+        this.searchUserCtrl.valueChanges
+            .pipe(
+                filter(value => {
+                    this.searchLoading = true;
+                    return typeof value === 'string';
+                }),
+                tap((value: string) => {
+                    if (value) {
+                        this.setTypingError();
+                    }
+                }),
+                debounceTime(500),
+                distinctUntilChanged(),
+                tap((value: string) => {
+                    if (value.trim()) {
+                        this.searchedValue = value;
+                    } else {
+                        this.searchUserCtrl.markAsPristine();
+                        this.searchUserCtrl.markAsUntouched();
+                    }
+                }),
+                tap(() => {
+                    this.resetSearchUsers();
+                }),
+                switchMap((searchTerm) => {
+                    let results$: Observable<IdentityUserModel[]> = of([]);
+                    if (this.appName) {
+                        results$ = this.peopleCloudService.findUsersBasedOnApp(this.clientId, this.roles, searchTerm);
+                    } else if (this.hasRoles()) {
+                        results$ = this.peopleCloudService.filterUsersBasedOnRoles(this.roles, searchTerm);
+                    } else {
+                        results$ = this.peopleCloudService.findUsers(searchTerm);
+                    }
+                    return results$;
+                }),
+                takeUntil(this.onDestroy$)
+            )
+            .subscribe((users) => {
+                this._searchUsers = users.filter((user) => !this.isUserAlreadySelected(user));
+                this.searchUsers$.next(this._searchUsers);
```

Comment:
from the failing e2e I see that the no results template is not present anymore. Can you double check if this refactoring affected it?

Reply:
Fixed failing tests.